### ISSUE TITLE
Angular 9 (Ivy) issue 

### ISSUE THREAD

**Issue by jongbonga** [nogorilla/angular-cc-library#61]
```
ERROR in node_modules/angular-cc-library/lib/directives.d.ts:1:22 - error NG6002: Appears in the NgModule.imports of CCPageModule, but could not be resolved to an NgModule class
```

```
Angular CLI: 9.0.1
Node: 10.16.0
OS: darwin x64

Angular: 9.0.0
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.25
@angular-devkit/build-angular     0.900.1
@angular-devkit/build-optimizer   0.900.1
@angular-devkit/build-webpack     0.900.1
@angular-devkit/core              9.0.1
@angular-devkit/schematics        9.0.1
@angular/cli                      9.0.1
@ngtools/webpack                  9.0.1
@schematics/angular               8.1.3
@schematics/update                0.900.1
rxjs                              6.5.4
typescript                        3.7.5
webpack                           4.41.2
```

**Comment by aserralta** [comment]
I have the same issue as @jongbonga

**Comment by antoine-souesme-needone** [comment]
Same issue here.

**Comment by DanWebb** [comment]
I think I'm seeing the same issue but with a different error message:

`Importing CreditCardDirectivesModule which does not have a ɵmod property`

`Unexpected value 'CreditCardDirectivesModule' imported by the module 'ControlCardNumberModule'. Please add an @NgModule annotation.`

I was able to fix it by manually re-creating [this module](https://github.com/nogorilla/angular-cc-library/blob/master/src/directives.ts)...:

```
import { NgModule } from '@angular/core';
import { CreditCardFormatDirective, ExpiryFormatDirective, CvcFormatDirective } from 'angular-cc-library';

const CREDIT_CARD_LIBRARY_DIRECTIVES = [
  CreditCardFormatDirective,
  ExpiryFormatDirective,
  CvcFormatDirective
];

@NgModule({
  declarations: [CREDIT_CARD_LIBRARY_DIRECTIVES],
  exports: [CREDIT_CARD_LIBRARY_DIRECTIVES]
})
export class CreditCardDirectivesModule {}

// Then...

@NgModule({
  imports: [CreditCardDirectivesModule, /* Other imports... */],
  // etc...
})
export class MyCustomModule { }
```

I guess it's some issue with the way the libraries built...?

**Comment by jongbonga** [comment]
@DanWebb are you able to run it with ivy enabled after your fix?

**Comment by DanWebb** [comment]
Sorry @jongbonga spoke too soon the above fails on a prod build in v9, it fooled me because it passed tests.

I'm sure it'll be something to do with how the libraries built or exported though.

**Comment by jongbonga** [comment]
I don’t want to get rid of this library since all the others are not satisfying my needs! But also, Can’t rock A9 without Ivy... it doesn’t make sense to me. 

Any other cc validation as good as this one? that compile with Ivy? Thanks for sharing 🙂

**Comment by DanWebb** [comment]
I've put some time into investigating and I think the issue is to do with the `.metadata.json` file not being in the root directory of the package https://unpkg.com/browse/angular-cc-library@1.2.5/

Anyway, there are lots of other similar occurrences coming up since the v9  release...

- https://github.com/angular/angular/issues/34983
- https://github.com/angular/angular/issues/35399
- https://github.com/angular/angular/issues/35411
...etc

I'm out of time and none of the listed solutions have worked so if nobody else has an idea I suppose it'll have to be left for a bit until enough people bring it up for an official solution to be released or an alternative lib is written...

**Comment by antoine-souesme-needone** [comment]
Yes, for now we can just upgrade angular to 9.x and disabled Ivy...

**Comment by rkajbaf** [comment]
It would be great to be able to use this with Ivy... for now I've removed this library but will gladly add it back into my project once this is fixed.

**Comment by antoine-souesme-needone** [comment]
Unfortunately this library does not seem to be maintained anymore..

**Comment by thehashrocket** [comment]
would it make sense to fork it and do the necessary updates?

**Comment by EfraimLA** [comment]
So we should leave this library for Ivy?

**Comment by thekip** [comment]
will be fixed when this PR gets merged https://github.com/nogorilla/angular-cc-library/pull/64

**Comment by EfraimLA** [comment]
Cool, so close issue?

**Comment by born2net** [comment]
same issue... fixed?
TX!

**Comment by thekip** [comment]
The PR is merged, and will be included in upcoming release with some other features

**Comment by born2net** [comment]
tx!

**Comment by jongbonga** [comment]
Thanks @thekip

**Comment by thekip** [comment]
Fixed in 2.0.0 https://github.com/nogorilla/angular-cc-library/releases/tag/v2.0.0
